test/boost/wasm_alloc_test.cc

```diff
@@ -98,6 +99,7 @@ SEASTAR_TEST_CASE(test_allocation_failures) {
         // Check that we can actually throw errors both during compilation and during execution
         BOOST_CHECK(errors_during_compilation > 0);
         BOOST_CHECK(errors_during_execution > 0);
+        alien_runner.stop();
```

Comment:
nit, might need to use `auto stop_alien_runner = deferred_stop(alien_runner);` right after it is defined? more idiomatic this way.

Reply:
Looks like the `deferred_stop` only works with `stop()` methods that return a future, so it can't be used easily here, but this stops being a problem anyway in the new approach.